config/v1/types_ingress.go

```diff
@@ -152,8 +153,53 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// Network Load Balancer. This field is present only if the type field is set to NLB.
+	//
+	// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+type AWSNetworkLoadBalancerParameters struct {
+	// eipAllocations assign Elastic IP addresses to the Network Load Balancer.
+	// The number of Allocation IDs must match the number of subnets that are used for the load balancer.
+	// See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html for general
+	// information about configuration, characteristics, and limitations of Elastic IP addresses.
+	// This field only specifies EIPs for the default IngressController. It is not a default value
+	// for all IngressControllers.
+	// +optional
+	// +listType=atomic
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="eipAllocations cannot contain duplicates"
+	// +kubebuilder:validation:MaxItems=10
+	EIPAllocations []EIPAllocation `json:"eipAllocations"`
+}
+
+// EIPAllocation is a reference to available eip allocations in AWS environment.
```

Comment:
Sorry @miheer, this was my suggestion. TIL.

Reply:
The content is good, it just needs to be written in prose so that it doesn't rely on rendering shapes like lists that work in markdown, but not in the openapi docs

Reply:
I am not able to access the link. However it looks like you want me to remove the hyphens.

Reply:
@JoelSpeed, I've been referring to https://github.com/openshift/api/pull/1730 as an example of how to format lists so that they render correctly in the generated files.  I hope I haven't been giving bad suggestions.  Is adding the extra line breaks insufficient?

Reply:
If that works then fine. I tend to go for prose but if the comment spacing can be made to work then I'm ok with that.

Reply:
line breaks don't work so we will need // in between two lines for next line